### ISSUE TITLE
libclang: crash detected during parsing:

### ISSUE THREAD

**Issue by kotoroshinoto** [rstudio/rstudio#5155]
https://github.com/rstudio/rstudio/issues/2268 might be related, but that was fixed, perhaps this is a regression, or something in my dev environment is broken?

### System details

    RStudio Edition : Desktop
    RStudio Version : 1.2.1335 Build 1379
    OS Version      :  Windows 7 Pro
    R Version       :  3.6.1

### Steps to reproduce the problem

edit and save any cpp file.

### Describe the problem in detail

libclang tries to analyze the files and crashes hard, spitting errors back through to the R terminal.

### Describe the behavior you expected

Whatever it typically does usually doesn't expose itself unless it identifies a clang related syntax suggestion or possible impropriety or error, which it would then highlight with red underlines or a line marker.

1. The output of sessionInfo()
R version 3.6.1 (2019-07-05)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 7 x64 (build 7601) Service Pack 1

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252  LC_CTYPE=English_United States.1252    LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                           LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
[1] compiler_3.6.1 tools_3.6.1    packrat_0.5.0 

2. The R code in question
(I don't think this is a code-generated bug, I think something is wrong in my environment, but for completeness, here is a link to the repo)
https://github.com/kotoroshinoto/Umediation

3. A diagnostics report; see https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report
[diagnostics-report.txt](https://github.com/rstudio/rstudio/files/3443350/diagnostics-report.txt)

I have been experiencing weird failures with devtools::install_github as well that I have been unable to reproduce on my other computers, so something might be very wrong with my user-environment on this install of windows. I have no idea how to diagnose the issue though.

The errors look like this:
```
Error in (function (command = NULL, args = character(), error_on_status = TRUE,  : 
  System command error
libclang: crash detected during parsing: {
  'source_filename' : 'C:/Users/mgooch/Desktop/repos/Umediation/src/shared_local_mediate_variables.cpp'
  'command_line_args' : ['clang', '-IC:/Program Files/RStudio/resources/libclang/builtin-headers/3.4', '-IC:/PROGRA~1/R/R-36~1.0/library/Rcpp/include', '-IC:/PROGRA~1/R/R-36~1.0/library/RCPPEI~1/include', '-std=gnu++11', '-IC:/PROGRA~1/R/R-36~1.0/include', '-DNDEBUG', '-include-pch', 'C:/Users/mgooch/AppData/Local/Temp/RtmpWKSqE8/rstudio/libclang/precompiled/Rcpp/59A35BDD/1.0.1-x86_64-w64-mingw32-76782-5.0.2/Rcpp-std=gnu++11.pch'],
  'unsaved_files' : [],
  'options' : 12,
}
libclang: crash detected during parsing: {
  'source_filename' : 'C:/Users/mgooch/Desktop/repos/Umediation/src/shared_local_mediate_variables.cpp'
  'command_line_args' : ['clang', '-IC:/Program Files/RStudio/resources/libclang/builtin-headers/3.4', '-IC:/PROGRA~1/R/R-36~1.0/library/Rcpp/include', '-IC:/PROGRA~1/R/R-36~1.0/library/RCPPEI~1/include', '-std=gnu++11', '-IC:/PROGRA~1/R/R-36~1.0/include', '-DNDEBUG', '-include-pch', 'C:/Users/mgooch/AppData/Local/Temp/RtmpWKSqE8/rstudio/libclang/precompiled/Rcpp/59A35BDD/1.0.1-x86_64-w64-mingw32-76782-5.0.2/Rcpp-std=gnu++11.pch'],
  'unsaved_files' : [],
  'options' : 2,
}
```

**Comment by kevinushey** [comment]
Is there anything special about that source file (`shared_local_mediate_variables.cpp`)? Maybe it contains some code construct that causes libclang to crash?

**Comment by kotoroshinoto** [comment]
I don't think so, it was working normally with pretty much the same code until recently.

I think something got borked in my dev environment, but i've got no idea how to start diagnosing what broke and how.

**Comment by ronblum** [comment]
@kotoroshinoto Does `devtools::install_github` work in standalone R, outside of RStudio, or does it fail there, as well?

**Comment by kotoroshinoto** [comment]
it fails for me on this system under all circumstances. Whether I call it from Rstudio, standalone R launched from the start menu, or an R terminal started in CMD.exe

I tried to reproduce the problem on the PC tower machine sitting next to my laptop, and it works normally there, so there is probably something about my laptop's environment interfering with the proper behavior of the install_github function.

I have no idea where to begin as far as figuring out what that is though.

**Comment by kotoroshinoto** [comment]
https://github.com/r-lib/remotes/issues/405

since the problems might be related, I figure it doesn't hurt to mention the issues across packages/repos.

**Comment by ronblum** [comment]
@kotoroshinoto I'm sorry it's not working on your laptop in anything running R, even though it works on the PC tower. This forum is meant for RStudio-specific issues, though. Two good places problems with R—like the one you've run into—are:
* [The RStudio Community](https://community.rstudio.com/)
* [Stack Overflow](https://stackoverflow.com/questions/tagged/r)

There are discussions between many R users in both places, and it's possible that someone has run into the same issues, or at least can provide some good guidance. Good luck!

**Comment by kotoroshinoto** [comment]
Is there a repo that would be able to look into the crashing of the libclang code parsing calls that are being generated from rstudio?

its probably not happening on the rstudio end, but i'm at a loss for how to deal with that or who to ask.

**Comment by ronblum** [comment]
@kotoroshinoto I'm sorry, I closed this based on `devtools::install_github` failing in R—which suggests that there's a more general issue—but I didn't address `libclang`, so I'm reopening.

1) You mentioned running `devtools::install_github` elsewhere. What about the cpp file—is there a problem on the other machine, too, or does it work there?
2) Do have this issue with any other .cpp files? For example, what happens with the default file? File -> New File -> C++ File
3) If the default file works fine, can you provide a simple example that results in the problem so that we can try to reproduce it? Thanks!

**Comment by kotoroshinoto** [comment]
it happens with any of the C++ files I edit as soon as the editor window tries to analyze the code.

However, it does NOT happen with a freshly created c++ file with the default Rcpp content.

I'd had issues in the past with code completion and analysis not quite working properly where the compiler was perfectly happy but RStudio was convinced that the types I was using didn't exist in their respective namespaces. (this was in the previous version of rstudio), and after I updated Rstudio and reinstalled rtools and R itself was updated from 3.6.0 to 3.6.1, I started to get this clang analyzer crash instead.

I am using eigen with Rcpp and a few of my files follow a more canonical C++ layout (paired header and implementation files). I've implemented structs and classes to make what i'm doing a bit easier and i'm leveraging threading (i am being careful with eigen in that regard, turning off eigen's own acceleration features, and letting the threads I spawn be the only multi-threading thats going on).

I think it MIGHT be the complexity of this code organization that is causing issues here, a few of the files include the various headers in multiple places as is per the norm in C++, but I suspect the assumptions made for the rstudio code analysis might be assuming a much simpler code organization than this.

All of my repos are using Similar codebase and includes though, so it still might be something in my code that is triggering this.
